Libraries/src/Amazon.Lambda.RuntimeSupport/bootstrap.sh

```diff
@@ -44,24 +51,25 @@ if [[ $(expr index "${LAMBDA_HANDLER}" ":") == 0 ]]; then
 
   DOTNET_ARGS+=("${EXECUTABLE_ASSEMBLY}")
 else
-  ASSEMBLY_NAME="${LAMBDA_DOTNET_MAIN_ASSEMBLY}"
-  if [ -z "${ASSEMBLY_NAME}" ]; then
-    DEPS_FILE=$(find "${USER_LAMBDA_BINARIES_DIR}" -name \*.deps.json -print)
-    if [ -z "${DEPS_FILE}" ]; then
-      echo "Error: .NET binaries for Lambda function are not correctly installed in the ${USER_LAMBDA_BINARIES_DIR} directory of the image when the image was built. The ${USER_LAMBDA_BINARIES_DIR} directory is missing the required .deps.json file." 1>&2
-      exit 1
-    fi
-    RUNTIMECONFIG_FILE=$(find "${USER_LAMBDA_BINARIES_DIR}" -name \*.runtimeconfig.json -print)
-    if [ -z "${RUNTIMECONFIG_FILE}" ]; then
-      echo "Error: .NET binaries for Lambda function are not correctly installed in the ${USER_LAMBDA_BINARIES_DIR} directory of the image when the image was built. The ${USER_LAMBDA_BINARIES_DIR} directory is missing the required .runtimeconfig.json file." 1>&2
-      exit 1
+  if [ -n "${LAMBDA_DOTNET_MAIN_ASSEMBLY}" ]; then
+    if [[ "${LAMBDA_DOTNET_MAIN_ASSEMBLY}" == *.dll ]]; then
+      ASSEMBLY_NAME="${LAMBDA_DOTNET_MAIN_ASSEMBLY::-4}"
+    else
+      ASSEMBLY_NAME="${LAMBDA_DOTNET_MAIN_ASSEMBLY}"
     fi
   else
-    if [[ "${ASSEMBLY_NAME}" == *.dll ]]; then
-      ASSEMBLY_NAME="${ASSEMBLY_NAME::-4}"
-    fi
-    DEPS_FILE="${USER_LAMBDA_BINARIES_DIR}/${ASSEMBLY_NAME}.deps.json"
-    RUNTIMECONFIG_FILE="${USER_LAMBDA_BINARIES_DIR}/${ASSEMBLY_NAME}.runtimeconfig.json"
+    ASSEMBLY_NAME="${LAMBDA_HANDLER::${HANDLER_COL_INDEX}-1}"
+  fi
+  
+  DEPS_FILE="${LAMBDA_TASK_ROOT}/${ASSEMBLY_NAME}.deps.json"
+  if ! [ -f "${DEPS_FILE}" ]; then
```

Comment:
This will not work if the function handler is referring to an assembly that is not the main assembly of the package bundle. This is a valid scenario in .NET Core 3.1 runtime and the previous `find` call allowed this to work.

We need the logic to do a fallback if this test fails to do a search in the top level directory for any other deps.json file and runtime.config.json file.

Reply:
Thanks @normj  - added a fallback that matches the .NET Core 3.1 RIC where the bootstrap would look for exactly one occurrence of `*.deps.json` and/or `*.runtimeconfig.json` in `LAMBDA_TASK_ROOT` and fail otherwise.